assets/src/vo/money.js

```diff
@@ -515,4 +516,45 @@ export default class Money {
 	static assertSameCurrency = ( currencyA, currencyB ) => {
 		assertSameCurrency( currencyA, currencyB );
 	}
+
+	/**
+	 * Receives an incoming value that could be a money formatted
+	 * string and returns a Money value object with the correct value
+	 * considering the provided currency.
+	 *
+	 * @param {string|number} moneyValue
+	 * @param {Currency} currency
+	 *
+	 * @return {Money} An instance of a money value object
+	 */
+	static fromMoneyValue = ( moneyValue, currency ) => {
+		assertCurrency( currency );
+		// detect if incoming value has a currency sign not matching provided
+		// currency.  This doesn't provide full protection from improper
+		// values sent in but is an initial safeguard.
+		if ( typeof moneyValue === 'string' ) {
+			const match = moneyValue.match( /[^\d\.\,\s]+/ );
+			if ( match && match[ 0 ] !== currency.sign ) {
```

Comment:
is this assuming the currency sign is the first character in the string?
what if the currency sign is at the end of the string?

Reply:
Nope, `match[ 0 ]` will be whatever gets returned by the regex expression (which will match the currency sign no matter where it is in the string).